massgov/mayflower-react
# DP-7578 Port Site Logo component

**Description**
This creates the Site Logo atom and adds it to the storybook. It also adds the Notes storybook addon to enable easily-viewable Markdown-based documentation.

A couple things were done differently with this component relative to previous components:
* A Markdown-formatted readme file is included for the component to document its intent/use and variables. This one is very short, as it's a straightforward component.
* Component data is imported from a JSON file, like the SocialLinks, but are passed in through the story, like [the Button component](https://github.com/massgov/mayflower-react/pull/2). This allows variants of the component to be used.
* The directory structure has been refined. All JS, JSON and readme files for a component are now grouped in the same directory. For the purposes of organization and consistency, existing categories/subcategories are reused from Mayflower, for example `/atoms/media/site-logo` in this case. In the near future, stories should also be broken down into component-specific files to help with organization and minimize merge conflicts.
* The component's directory structure is mimicked in Storybook, in this instance going under Atoms > Media > Site Logo > Site Logo. While this may seem like overkill at this point (particularly having a single Site Logo component in the Site Logo category), it should help keep things organized when component variants are needed.

**Jira**
https://jira.state.ma.us/browse/DP-7578

**To Test**
[ ] Pull the branch down and run `npm install` and then `npm start`
[ ] Navigate to http://localhost:6006/ and click through Atoms > Media > Site Logo > Site Logo
[ ] Confirm that the logo component looks like the screenshot below and the [Mayflower equivalent](https://mayflower.digital.mass.gov/?p=atoms-site-logo).
[ ] Click on the "Notes" tab below the component preview and confirm that a readme is displayed and has (minimal) Markdown formatting applied.

**Screenshot**
<img width="700" src="https://user-images.githubusercontent.com/18170074/35409829-6010b2a6-01e1-11e8-91ea-2b9003d4e1bf.png">
